example/storage-provider-secrets/00-azure-blob-storage-secret.yaml

```diff
@@ -7,6 +7,7 @@ type: Opaque
 data: 
   storageAccount: YWRtaW4= # admin
   storageKey: YWRtaW4= # admin
+  # storageAPIEndpoint: http[s]://<storage-account>.<domain> # override the ABS endpoint
```

Comment:
Voting for `storageEndpoint`

Reply:
We'd decided to name it `storageAPIEndpoint` since that is what the override is named currently for GCS.

Would you suggest we break away from consistency and go with this?

Reply:
I did not know about GCS approach, but on a local level next to `storageKey` it's less "locally consistent"

Reply:
Yes, you are right. It doesn't look super coherent in that sense.

I'll wait for @unmarshall to chime in with his opinions before we decide on what to go ahead with.

Reply:
If it's used by the Azurite emulator with the same name already, then it's okay - no need to make this harder.

Reply:
This isn't making it harder really, the emulator field was only intended for developers so we aren't breaking any public APIs.

It's better to have a more coherent scheme now like you mentioned, and we shouldn't refrain from changing this up now in my opinion.

Reply:
Well, we have two choices - either use the same field name `storageAPIEndpoint` for all providers, since that provides consistency for users who use EtcdBR for multiple providers. But on the other hand, since each provider specifies its own fields already, we are already "inconsistent", in the sense that each provider already has its own set of credential fields. So using a local name for the endpoint override does make sense in this context. If this is the case, then we need to make the change for both Azure as well as GCS (since GCS uses the term `regionalEndpoint` rather than `storageAPIEndpoint`). And since v0.29.0 is already released and the field name `storageAPIEndpoint` in the GCS secret is already being used by users, it would take time and effort to roll out this change and eventually enforce it.

Reply:
@kon-angelo in any provider specific configuration we have used names which are also similar/same to the ones that are used by the respective provider in the APIs that they expose. This establishes a 1:1 correspondence with what the infra provider then provides.  

Since these secrets are already very specific to a provider. E.g. for AWS you will have accessKeyID and secretAccessKey whereas for azure you will have storageAccount and storageKey. These names are completely in line with what the provider wants. One could extend this argument and say that the endpoint that is required to connect to the object store should also follow the naming convention as defined by the provider in order to be consistent. 

From this argument using [serviceURL](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/service/client.go#L70C8-L70C17) should be sufficient. Unfortunately this is just the name of an argument to a constructor function and cannot be construed strictly as an API field name.

Similarly in GCS they just call it [Endpoint](https://github.com/googleapis/google-api-go-client/blob/v0.189.0/option/option.go#L74).

If you have already released it in v0.29.0 then we are now stuck with `storageAPIEndpoint` - i agree with @kon-angelo  that `API` is not really required in the name. Then lets stick to using one name for now and when we refactor backup-restore then we can break the API then.

Reply:
Currently v0.29.0 is not being used by gardener. In addition if you have a look at [ConstructBlobServiceURL](https://github.com/gardener/etcd-backup-restore/blob/v0.29.0/pkg/snapstore/abs_snapstore.go#L85-L107) then only to support emulator we use an environment variable [AZURE_STORAGE_API_ENDPOINT](https://github.com/gardener/etcd-backup-restore/blob/v0.29.0/pkg/snapstore/abs_snapstore.go#L33). For productive use cases we do not use it.

So we have an option to change it in v0.28.x and then adapt the changes in v0.29.x without too much issues w.r.t backward compatibility since no productive use cases are impacted.

Reply:
Please check [b154104](https://github.com/gardener/etcd-backup-restore/pull/756/commits/b15410461787025718d22ad3ca4823a6ca012970).